src/nc/core/image/Image.cpp

```diff
@@ -108,6 +108,13 @@ void Image::setDemangler(std::unique_ptr<mangling::Demangler> demangler) {
     demangler_ = std::move(demangler);
 }
 
+void Image::setEntryPoint(ByteAddr address) {
+    auto section = getSectionContainingAddress(address);
+
+    if (section && section->isExecutable()) {
```

Comment:
Didn't consider the case where the section isn't added yet. Good point, removed the check.